src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -980,15 +980,16 @@ void InstanceKlass::add_initialization_error(JavaThread* current, Handle excepti
   // If the initialization error is OOM, this might not work, but if GC kicks in
   // this would be still be helpful.
   JavaThread* THREAD = current;
-  Handle cause = java_lang_Throwable::get_cause_with_stack_trace(exception, THREAD);
-  if (HAS_PENDING_EXCEPTION || cause.is_null()) {
-    CLEAR_PENDING_EXCEPTION;
+  Handle init_error = java_lang_Throwable::create_initialization_error(current, exception);
+
+  if ( init_error.is_null()) {
+    log_trace(class, init)("Initialization error is null for class %s", external_name());
```

Comment:
You need to move the ResourceMark from line 995 to line 984 to cover this new logging statement.